pkg/ingester/activeseries/active_series.go

```diff
@@ -87,10 +87,10 @@ func (c *ActiveSeries) CurrentConfig() CustomTrackersConfig {
 }
 
 // UpdateSeries updates series timestamp to 'now'. Function is called to make a copy of labels if entry doesn't exist yet.
-func (c *ActiveSeries) UpdateSeries(series labels.Labels, fp uint64, now time.Time, labelsCopy func(labels.Labels) labels.Labels) {
-	stripeID := fp % numStripes
+func (c *ActiveSeries) UpdateSeries(series labels.Labels, ref uint64, now time.Time, labelsCopy func(labels.Labels) labels.Labels) {
+	stripeID := ref % numStripes
```

Comment:
would this result is more uneven distribution among the stripes? I wonder if we still shouldn't hash the ref to choose a stripe ID

Reply:
Good question, I was hoping to keep this as fast as possible since it's called for every sample ingested. Do you have a suggestion for a hash function here? 

Series refs are generated by incrementing a uint64, so in the case of 0 churn, a simple modulo would have ~perfect distribution. Of course, in reality there's always churn. 😱

Reply:
I wrote a small program to simulate 100k series with 40% random series churn, and found the following for the bucket sizes:

| hash function | min | max | avg |
|---------------|-----|-----|-----|
| mod           | 168 | 222 | 195 |
| fnv1a         | 166 | 225 | 195 |
| fnv           | 168 | 223 | 195 |
| crc32 ieee    | 171 | 223 | 195 |

Based on this, it seems that mod is essentially as good as other hash functions in this case, while also being simpler to understand IMO.